cocotb/binary.py

```diff
@@ -295,22 +313,19 @@ def get_value_signed(self):
         else:
             return -1 * (1 + (int(~ival) & (signbit - 1)))
 
-    def set_value(self, integer):
-        self._str = self._convert_to[self.binaryRepresentation](integer)
+    get_value_signed = signed_integer.fget
 
     @property
     def is_resolvable(self):
         """Does the value contain any ``X``'s?  Inquiring minds want to know."""
         return not any(char in self._str for char in BinaryValue._resolve_to_error)
 
-    value = property(get_value, set_value, None,
-                     "Integer access to the value. **deprecated**")
-    integer = property(get_value, set_value, None,
-                       "The integer representation of the underlying vector.")
-    signed_integer = property(get_value_signed, set_value, None,
-                              "The signed integer representation of the underlying vector.")
+    @signed_integer.setter
+    def signed_integer(self, val):
+        self.integer = val
```

Comment:
Might make sense to put this on line 315 so that the getter and setter are adjacent.